pkg/acquisition/modules/kinesis/kinesis.go

```
@@ -0,0 +1,473 @@
+		sess = session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+			Profile:           *k.Config.AwsProfile,
+		}))
+	} else {
+		sess = session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+	}
+
+	if sess == nil {
+		return fmt.Errorf("failed to create aws session")
+	}
+	config := aws.NewConfig()
+	if k.Config.AwsRegion != "" {
+		config = config.WithRegion(k.Config.AwsRegion)
+	}
+	if k.Config.AwsEndpoint != "" {
+		config = config.WithEndpoint(k.Config.AwsEndpoint)
+	}
+	k.kClient = kinesis.New(sess, config)
+	if k.kClient == nil {
+		return fmt.Errorf("failed to create kinesis client")
+	}
+	return nil
+}
+
+func (k *KinesisSource) GetMetrics() []prometheus.Collector {
+	return nil
+}
+func (k *KinesisSource) GetAggregMetrics() []prometheus.Collector {
+	return nil
+}
+
+func (k *KinesisSource) Configure(yamlConfig []byte, logger *log.Entry) error {
+	config := KinesisConfiguration{}
+	k.logger = logger
+	err := yaml.UnmarshalStrict(yamlConfig, &config)
+	if err != nil {
+		return errors.Wrap(err, "Cannot parse kinesis datasource configuration")
+	}
+	if config.Mode == "" {
+		config.Mode = configuration.TAIL_MODE
+	}
+	k.Config = config
+	if k.Config.StreamName == "" && !k.Config.UseEnhancedFanOut {
+		return fmt.Errorf("stream_name is mandatory when use_enhanced_fanout is false")
+	}
+	if k.Config.StreamARN == "" && k.Config.UseEnhancedFanOut {
+		return fmt.Errorf("stream_arn is mandatory when use_enhanced_fanout is true")
+	}
+	if k.Config.ConsumerName == "" && k.Config.UseEnhancedFanOut {
+		return fmt.Errorf("consumer_name is mandatory when use_enhanced_fanout is true")
+	}
+	err = k.newClient()
+	if err != nil {
+		return errors.Wrap(err, "Cannot create kinesis client")
+	}
+	k.shardReaderTomb = &tomb.Tomb{}
+	return nil
+}
+
+func (k *KinesisSource) ConfigureByDSN(string, map[string]string, *log.Entry) error {
+	return nil
```

do we want to return an error if we don't support dsn ?